helm_deploy/templates/configuration.yaml

```diff
@@ -0,0 +1,10 @@
+{{- if not (lookup "v1" "ConfigMap" .Release.Namespace (include "laa-submit-crime-forms.fullname" .)) -}}
```

Comment:
Can you explain what this line means? Is this a "if this isn't a branch" check?

Reply:
having this template in our helm means that it will get recreated on every deployment - this includes re-establishing the default value of maintenance mode = false - we don't want that to happen as we want whatever we've already set on the configmap to take presedence - we only want the configmap to get recreated if it's been deleted for whatever reason

This checks if the configmap exists by doing a lookup of the version, kind, namespace and name and will only generate the configmap if it doesn't already exist

Reply:
Ah. So the readme says that what's preventing it from being reset on every deployment is that it "has a helm.sh/resource-policy of keep". Does the "if" clause here make the resource-policy redundant, and if so is it worth updating the readme?

Reply:
oh sorry the readme/this doesn't fully explain the situation: 
1. We don't want the config map to get reset to defaults --> only generate if doesn't exist because generating it again resets the values
2. We don't want the config map to get deleted --> because the if statement removes the template from the deployment, when it's false, the configmap doesn't get left alone, it gets deleted - therefore we have the resource policy to ensure it doesn't get deleted

I should probably be more clear about this in the readme as another caveat is that if we want to make changes for example to the metadata of the resource, we would also have to delete the existing resource to allow the new changes to come into effect

Reply:
I would have preferred a policy that doesn't reset the data to simplify things but I couldn't find anything
only other option I could think of is to keep the data of the configmap empty, default to false in the code and only use the configmap when data is present - however that would mean awkwardly adding a fallback to the helm deployment when bringing in the environment variable and settings being less explicit as to where they come from